src/main/java/org/primefaces/component/cascadeselect/CascadeSelectRenderer.java

```diff
@@ -247,6 +256,36 @@ protected void encodeListItems(FacesContext context, CascadeSelect cascadeSelect
         }
     }
 
+    /**
+     * Recursive method used to find a SelectItem by its value.
+     * @param context FacesContext
+     * @param selectItems the List of SelectItems
+     * @param value the input value to search for
+     * @return either the SelectItem found or NULL if not found
+     */
+    protected SelectItem findItemByValue(FacesContext context, List<SelectItem> selectItems, String value) {
+        SelectItem foundValue = null;
+        for (int i = 0; i < selectItems.size(); i++) {
+            SelectItem item = selectItems.get(i);
+            if (item instanceof SelectItemGroup) {
+                SelectItemGroup selectItemGroup = (SelectItemGroup) item;
+                foundValue = findItemByValue(context, Arrays.asList(selectItemGroup.getSelectItems()), value);
+                if (foundValue != null) {
+                    break;
+                }
+            }
+            else {
+                String itemValue = Objects.toString(item.getValue());
+                if (value.equalsIgnoreCase(itemValue)) {
```

Comment:
this will lead to a NPE, when value==null
i would just use Object.equals, without ignoreCase